code/game/objects/structures/catwalk.dm

```diff
@@ -3,47 +3,25 @@
 	desc = "Cats really don't like these things."
 	icon = 'icons/obj/catwalks.dmi'
 	icon_state = "catwalk"
-	density = 0
-	anchored = 1.0
-	var/obj/item/stack/tile/mono/plated_tile
-
+	density = FALSE
+	anchored = TRUE
 	layer = CATWALK_LAYER
-	var/hatch_open = FALSE
-	footstep_sounds= list(
+	footstep_sounds = list(
 		'sound/effects/footstep/catwalk1.ogg',
 		'sound/effects/footstep/catwalk2.ogg',
 		'sound/effects/footstep/catwalk3.ogg',
 		'sound/effects/footstep/catwalk4.ogg',
 		'sound/effects/footstep/catwalk5.ogg')
-
-	can_block_movement = FALSE //It IS the floor
-	atom_flags = ATOM_FLAG_UNTARGETABLE
-
-/obj/structure/catwalk/register_zstructure(var/turf/T)
-	LAZYSET(T.zstructures, src, 1)	//Ladders have a ztransition priority of 2 to overrule other things
+	obj_flags = OBJ_FLAG_NOFALL
+	var/hatch_open = FALSE
+	var/obj/item/stack/tile/mono/plated_tile
 
 /obj/structure/catwalk/Initialize()
 	. = ..()
-	for(var/obj/structure/catwalk/C in get_turf(src))
-		if(C != src)
-			qdel(C)
-	var/turf/T = get_turf(src)
-	register_zstructure(T)
```

Comment:
this zstructure feature is an addition of mine, to optimise movement. 

Before i added it, things used to check every object in the tile to see if they could prevent falling, now they only check things in the zstructures list

Removing it is fine if you've duplicated the functionality elsewhere, but i don't want to see the bad method returning